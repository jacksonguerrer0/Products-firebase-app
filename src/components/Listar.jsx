import React, { useEffect } from 'react'
import { Table } from 'react-bootstrap'
import {  useDispatch, useSelector } from 'react-redux'
import { listarProducto } from '../redux/productsDuck'

const Listar = () => {
    const dispatch = useDispatch()
    const {productos} = useSelector(store => store.producto)
   // const { productos } = useSelector(store => store.producto)
    // const producto = useSelector(store => store)
    useEffect(() => {
        dispatch(listarProducto())
    }, [dispatch])

    return (
        <div>
            <button onClick={()=> dispatch(listarProducto())}>Listar Producto</button>
            <h1>Productos CampoStore</h1>
            <Table striped bordered hover>
                <thead>
                    <tr>
                        <th>Id</th>
                        <th>Nombre</th>
                        <th>Precio</th>
                        <th>Vendedor</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        
                        productos && productos.map(ele => (
                            <tr key={ele.id + ele.nombre + ele.precio + ele.vendedor}>
                                <td>{ele.id}</td>
                                <td>{ele.nombre}</td>
                                <td>{ele.precio}</td>
                                <td>{ele.vendedor}</td>
                            </tr>
                        ))
                    }
                </tbody>
            </Table>
        </div>
    )
}

export default Listar
