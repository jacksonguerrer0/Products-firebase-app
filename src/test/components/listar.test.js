import React from 'react'
import { shallow } from 'enzyme'
import Listar, { cantidad } from '../../components/Listar'
// import { useSelector } from 'react-redux'

describe('Test <Listar />', () => {

    test('Comprobar === longitud array card', () => {
        const card = [
            {
            id: 1,
            url: 'https://helpx.adobe.com/content/dam/help/en/photoshop/using/convert-color-image-black-white/jcr_content/main-pars/before_and_after/image-before/Landscape-Color.jpg',
            precio: '23422',
            nombre: 'Pi√±a', 
            vendedor: 'Jackson', 
            },
            {
                id: 2,
                url: 'https://helpx.adobe.com/content/dam/help/en/photoshop/using/convert-color-image-black-white/jcr_content/main-pars/before_and_after/image-before/Landscape-Color.jpg',
                precio: '23422',
                nombre: 'Sandia', 
                vendedor: 'Jorge', 
            }
        ]
        const wrapper = shallow(<Listar defaultCard={card} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find(`.product`).length).toBe(card.length)
    })
    test('Validar titulo', () => {
        expect(cantidad()).toEqual({nombre: 'jackson'})
    })
    
})
