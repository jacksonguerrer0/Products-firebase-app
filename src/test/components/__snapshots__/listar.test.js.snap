// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Listar /> Comprobar === longitud array card 1`] = `
<div>
  <h1>
    Productos CampoStore
  </h1>
  <ForwardRef
    bordered={true}
    hover={true}
    striped={true}
  >
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Imagen
        </th>
        <th>
          Nombre
        </th>
        <th>
          Precio
        </th>
        <th>
          Vendedor
        </th>
        <th>
          Accion
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="product"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          <img
            alt=""
            src="https://helpx.adobe.com/content/dam/help/en/photoshop/using/convert-color-image-black-white/jcr_content/main-pars/before_and_after/image-before/Landscape-Color.jpg"
            style={
              Object {
                "width": "100px",
              }
            }
          />
        </td>
        <td>
          Pi√±a
        </td>
        <td>
          23422
        </td>
        <td>
          Jackson
        </td>
        <td>
          <button>
            Editar
          </button>
          <button>
            Eliminar
          </button>
        </td>
      </tr>
      <tr
        className="product"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          <img
            alt=""
            src="https://helpx.adobe.com/content/dam/help/en/photoshop/using/convert-color-image-black-white/jcr_content/main-pars/before_and_after/image-before/Landscape-Color.jpg"
            style={
              Object {
                "width": "100px",
              }
            }
          />
        </td>
        <td>
          Sandia
        </td>
        <td>
          23422
        </td>
        <td>
          Jorge
        </td>
        <td>
          <button>
            Editar
          </button>
          <button>
            Eliminar
          </button>
        </td>
      </tr>
    </tbody>
  </ForwardRef>
</div>
`;
